digdag-standards/src/main/java/io/digdag/standards/operator/td/TDJobOperator.java

```diff
@@ -7,59 +7,92 @@
 import com.google.common.base.Throwables;
 import com.google.common.collect.ImmutableList;
 import com.treasuredata.client.TDClient;
+import com.treasuredata.client.TDClientHttpException;
 import com.treasuredata.client.model.TDJob;
 import com.treasuredata.client.model.TDJobSummary;
 import com.treasuredata.client.model.TDResultFormat;
+import io.digdag.spi.SecretProvider;
 import io.digdag.spi.TaskExecutionException;
+import io.digdag.util.RetryExecutor;
 import io.digdag.util.RetryExecutor.RetryGiveupException;
 import org.msgpack.core.MessagePack;
 import org.msgpack.core.MessageUnpacker;
 import org.msgpack.value.ArrayValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.util.Iterator;
 import java.util.List;
+import java.util.concurrent.Callable;
 import java.util.zip.GZIPInputStream;
 
+import static io.digdag.standards.operator.td.TDOperator.AUTH_MAX_RETRY_LIMIT;
 import static io.digdag.standards.operator.td.TDOperator.defaultRetryExecutor;
+import static io.digdag.standards.operator.td.TDOperator.isAuthenticationErrorException;
+import static io.digdag.util.RetryExecutor.retryExecutor;
 
 class TDJobOperator
 {
-    private final TDClient client;
+    private TDClient client;
     private final String jobId;
+    private final SecretProvider secrets;
 
-    TDJobOperator(TDClient client, String jobId)
+    private static final Logger logger = LoggerFactory.getLogger(TDJobOperator.class);
+
+    TDJobOperator(TDClient client, String jobId, SecretProvider secrets)
     {
         this.client = client;
         this.jobId = jobId;
+        this.secrets = secrets;
+    }
+
+    void updateApikey(SecretProvider secrets)
+    {
+        String apikey = TDClientFactory.getApikey(secrets);
+        client = client.withApiKey(apikey);
+    }
+
+    RetryExecutor authenticatinRetryExecutor() {
```

Comment:
This method is similar to TDOperator.authenticatinRetryExecutor().
But some options are not set.
Could you explain difference between them ?
Can't we merge and reuse same logic?

Reply:
`updateApikey` update client member in each TDOperator and TDJobOperator, and td-client doesn't have overwrite method for itself, and there is only `withApiKey` which returns new client, I thought I have to update each client member in each class, so I couldn't merge this method with passing client.
Sorry if I'm wrong.

Reply:
Got it.  There is a possibility to fix by using base class. But anyway I understand.